plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -2770,6 +2811,25 @@ private void mountVmfsDatastore(DatastoreMO dsMO, List<Pair<ManagedObjectReferen
         }
     }
 
+    private void mountVmfsDatastore2(DatastoreMO dsMO, List<HostMO> hosts) throws Exception {
+        for (HostMO hostMO : hosts) {
+            if (!isDatastoreMounted(dsMO, hostMO)) {
+                HostStorageSystemMO hostStorageSystemMO = hostMO.getHostStorageSystemMO();
+
+                try {
+                    hostStorageSystemMO.mountVmfsVolume(getDatastoreUuid(dsMO, hostMO));
+                }
+                catch (InvalidStateFaultMsg ex) {
+                    List<HostMO> currentHosts = new ArrayList<>(1);
```

Comment:
In these cases when there is an exception captured, I think you can log a message in TRACE or DEBUG to say that an exception of type X has been caught, and this triggered some processing. In this case, the "InvalidStateFaultMsg" is captured and this triggers a code to wait until the datastores are mounted. The message can be logged in DEBUG and the exception in TRACE level.